pkg/fileutils/fileutils.go

```diff
@@ -129,16 +137,34 @@ func (pm *PatternMatcher) MatchesResult(file string) (res *MatchResult, err erro
 
 		if match {
 			res.isMatched = !negative
-			if negative {
-				res.excludes++
-			} else {
+			if res.isMatched {
 				res.matches++
+				matchPatterns = append(matchPatterns, pattern)
+			} else {
+				res.excludes++
 			}
 		}
 	}
 
-	if res.matches > 0 {
+	if res.isMatched {
```

Comment:
The previous implementation had a bug. When the rule is `["1", "!1"]`, it would also enter this conditional statement, but it should not.